pkg/restapi/v1/verifier/controller.go

```diff
@@ -350,6 +361,57 @@ func (c *Controller) initiateOidcInteraction(
 	}, err
 }
 
+func applyPresentationDefinitionFilters(pd *presexch.PresentationDefinition,
+	filters *PresentationDefinitionFilters) (*presexch.PresentationDefinition, error) {
+	return applyFieldsFilter(pd, lo.FromPtr(filters.Fields))
+}
+
+func applyFieldsFilter(pd *presexch.PresentationDefinition, fields []string) (*presexch.PresentationDefinition, error) {
+	for _, desc := range pd.InputDescriptors {
+		var filteredFields []*presexch.Field
+
+		for _, field := range desc.Constraints.Fields {
+			matched, err := matchField(fields, field.ID)
+			if err != nil {
+				return nil, err
+			}
+
+			if matched {
+				filteredFields = append(filteredFields, field)
+			}
+		}
+
+		desc.Constraints.Fields = filteredFields
+	}
+
+	return pd, nil
+}
+
+func matchField(ids []string, target string) (bool, error) {
+	const wildcard = "*"
+
+	for _, id := range ids {
+		// this case covers both exact id and empty string rule
+		if id == target {
+			return true, nil
+		}
+
+		if strings.Contains(id, wildcard) {
+			exp := strings.ReplaceAll(id, wildcard, "."+wildcard)
+			r, err := regexp.Compile(exp)
+			if err != nil {
+				return false, fmt.Errorf("failed to compile regex=%s. %w", exp, err)
+			}
+
+			if r.MatchString(target) {
+				return true, nil
+			}
+		}
```

Comment:
I think, if id has any special characters we may need to escape it.

Reply:
otherwise regex compile might break

Reply:
I thought it was supposed to break it - I have a test that breaks it. Do we support special chars in id?